src/components/views/auth/InteractiveAuthEntryComponents.tsx

```diff
@@ -440,11 +481,54 @@ export class EmailIdentityAuthEntry extends React.Component<IEmailIdentityAuthEn
         } else {
             return (
                 <div className="mx_InteractiveAuthEntryComponents_emailWrapper">
-                    <p>{ _t("A confirmation email has been sent to %(emailAddress)s",
+                    <AuthHeader
+                        title={_t("Check your email to continue")}
+                        icon={<img
+                            src={EmailPromptIcon}
+                            alt={_t("Unread email icon")}
+                            width={16}
+                        />}
+                        hideServerPicker={true}
+                    />
+                    <p>{ _t("To create your account, open the link in the email we just sent to %(emailAddress)s.",
                         { emailAddress: <b>{ this.props.inputs.emailAddress }</b> },
+                    ) }</p>
+                    { this.state.requesting ? (
+                        <p className="secondary">{ _t("Did not receive it? <a>Resend it</a>", {}, {
+                            a: (text: string) => <Fragment>
+                                <AccessibleButton
+                                    kind='link_inline'
+                                    onClick={() => null}
+                                    disabled
+                                >{ text }</AccessibleButton>
+                                <Spinner />
+                            </Fragment>,
+                        }) }</p>
+                    ) : this.state.requested ? (
+                        <p className="secondary">{ _t("Did not receive it? <a>Requested</a>", {}, {
+                            a: (text: string) => <AccessibleButton
+                                kind='link_inline'
+                                onClick={() => null}
+                                disabled
+                            >{ text }</AccessibleButton>,
+                        }) }</p>
+                    ) : (
+                        <p className="secondary">{ _t("Did not receive it? <a>Resend it</a>", {}, {
+                            a: (text: string) => <AccessibleButton
+                                kind='link_inline'
+                                onClick={async () => {
+                                    this.setState({ requesting: true });
+                                    try {
+                                        await this.props.requestEmailToken();
```

Comment:
requestEmailToken is optional, are we really relying on a try-catch here to handle the unset case?

Reply:
The try-catch is for handling the "requesting the token fails" case, but good point!